src/purifier/sanitation.js

```diff
@@ -264,6 +264,7 @@ const DENYLISTED_TAG_SPECIFIC_ATTRS = Object.freeze(
     'input': DENYLISTED_FIELDS_ATTR,
     'textarea': DENYLISTED_FIELDS_ATTR,
     'select': DENYLISTED_FIELDS_ATTR,
+    'template': ['shadowroot'],
```

Comment:
Is sanitization of the template even necessary? Once the template is parsed by the parser, the `<template>`'s contents are cleared out to create the shadow root. Then there are two possiblities:

1. We take the `innerHTML`, which means the template will just be empty and shadow root will not be included
2. We import the real DOM node, in which case the node will be inactive (it's part of another _other_ document)

I think we only need to guard the server, right?

Reply:
I think the protection here is with two templates:

```
<template type="amp-mustache">
  <div>
    <template shadowroot>do something horrible here</template>
  </div>
</template>
```

Reply:
i believe nested templates are already banned: https://amp.dev/documentation/components/amp-mustache/#nested-templates

Reply:
Thanks a lot, @samouri. That covers it. I'm closing this pull request then.